FHICORC.Core/Services/Model/SmartHealthCardModel/Shc/CredentialSubject.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using FHICORC.Core.Services.Enum;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace FHICORC.Core.Services.Model.SmartHealthCardModel.Shc
+{
+    public class CredentialSubject
+    {
+        [JsonConstructor]
+        public CredentialSubject(string FhirVersion, JObject FhirBundle)
+        {
+            Immunizations = new List<SmartHealthCardImmunization>();
+            Observations = new List<SmartHealthCardObservation>();
+            Patients = new List<SmartHealthCardPatient>();
+
+            this.FhirVersion = FhirVersion;
+            JArray entries = (JArray)FhirBundle["entry"];
+
+            foreach (JObject entry in entries)
+            {
+                string resourceType = (string)entry["resource"]["resourceType"];
+                if (System.Enum.TryParse(resourceType, out ResourceType type))
+                {
+                    switch (type)
+                    {
+                        case ResourceType.Patient:
+                            var patient = JsonConvert.DeserializeObject<SmartHealthCardPatient>(entry["resource"].ToString());
+                            Patients.Add(patient);
+                            break;
+                        case ResourceType.Immunization:
+                            var immunization = JsonConvert.DeserializeObject<SmartHealthCardImmunization>(entry["resource"].ToString());
+                            Immunizations.Add(immunization);
+                            break;
+                        case ResourceType.Observation:
+                            var observation = JsonConvert.DeserializeObject<SmartHealthCardObservation>(entry["resource"].ToString());
+                            Observations.Add(observation);
+                            break;
```

Comment:
Don't we need some default for this switch statement in case the entry is neither of these? For instance, throw an exception or something.

Reply:
If type is not handled (TryParse fails or switch is not exhausted) it should just be skipped (like here) I think. 
So we only care for the data we need and handle the potential scenario where the card also contains more data types. Also a test to verify this works as expected: _CredentialSubject_UnknownType_DoesNotThrowException_